pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -0,0 +1,572 @@
+          this.toggleError('duplicate', true, INPUT.edit);
+
+          return;
+        }
+
+        this.updateItems(items);
+      }
+      this.toggleEditMode(false);
+    },
+
+    /**
+     * Delete item and update items list
+     */
+    deleteItem(item?: string): number {
+      let index = -1;
+
+      if (item) {
+        index = findStringIndex(this.items as string[], item, false);
+        const items = this.items.filter(f => f !== item) as string[];
+
+        this.updateItems(items);
+      }
+
+      return index;
+    },
+
+    /**
+     * Update items list and emit to parent component
+     */
+    updateItems(items: string[]) {
+      this.$emit('change', items);
+    },
+  },
+
+});
+</script>
+
+<template>
+  <div
+    class="string-list"
+    :class="{ readonly }"
+  >
+    <div
+      ref="box"
+      data-testid="div-string-list-box"
+      class="string-list-box"
+      tabindex="0"
+      @dblclick="onClickEmptyBody()"
+    >
+      <div
+        v-for="item in items"
+        :key="item"
+        :ref="item"
+        :class="{
+          selected: selected === item,
+          readonly
+        }"
+        :data-testid="`div-item-${item}`"
+        class="item static"
+        tabindex="0"
+        @mousedown="onSelect(item)"
+        @dblclick.stop="toggleEditMode(true, item)"
+        @keydown.down.prevent="onSelectNext('down')"
+        @keydown.up.prevent="onSelectNext('up')"
+        @blur="onSelectLeave(item)"
+      >
+        <span
+          v-if="!isEditItem || isEditItem !== item"
+          class="label static"
+        >
+          {{ item }}
+        </span>
+        <input
+          v-if="isEditItem && isEditItem === item"
+          ref="item-edit"
+          class="input-field edit-input static"
+          :value="value != null ? value : item"
+          @input.prevent="onChange($event.target.value)"
+          @blur.prevent="toggleEditMode(false)"
+          @keydown.enter.prevent="updateItem(item)"
+        />
+      </div>
+      <div
+        v-if="isCreateItem"
+        class="create-field static"
+      >
+        <input
+          ref="item-create"
+          type="text"
+          class="input-field create-input static"
+          :placeholder="placeholder"
+          @input.prevent="onChange($event.target.value)"
+          @keydown.enter.prevent="saveItem"
+        />
+      </div>
+    </div>
+    <div
+      v-if="!readonly"
+      class="string-list-footer"
+      :class="{[actionsPosition]: true }"
+    >
+      <div
+        data-testid="div-action-buttons"
+        class="action-buttons"
+      >  
+        <button
+          class="btn btn-sm role-tertiary add-button"
+          :disabled="!selected && !isCreateItem && !isEditItem"
+          @mousedown.prevent="onClickMinusButton"
+        >
+          <span class="icon icon-minus icon-sm" />
+        </button>
+        <button
+          class="btn btn-sm role-tertiary remove-button"
+          :disabled="isCreateItem"
```

Comment:
Remove button (-) should be set disabled if nothing is selected.

Reply:
I have onfused remove with add button :face_with_spiral_eyes: